<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>847</width>
    <height>540</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Data Processor</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <widget class="QTabWidget" name="tabWidget_2">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="tab">
         <attribute name="title">
          <string>数据文件发送</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QTabWidget" name="tabWidget">
              <property name="currentIndex">
               <number>0</number>
              </property>
              <widget class="QWidget" name="basicTab">
               <attribute name="title">
                <string>基本设置</string>
               </attribute>
               <layout class="QGridLayout" name="gridLayout_9">
                <item row="0" column="0">
                 <layout class="QHBoxLayout" name="horizontalLayout">
                  <item>
                   <widget class="QLineEdit" name="dirEdit">
                    <property name="placeholderText">
                     <string>默认为程序运行目录, 不可为空</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QPushButton" name="browseButton">
                    <property name="text">
                     <string>浏览...</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item row="1" column="0">
                 <layout class="QHBoxLayout" name="horizontalLayout_2">
                  <item>
                   <widget class="QLineEdit" name="addrEdit">
                    <property name="placeholderText">
                     <string>输入要添加的地址和端口后点击右侧按钮添加, 格式IP:Port</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QPushButton" name="addAddrButton">
                    <property name="text">
                     <string>添加</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QPushButton" name="removeAddrButton">
                    <property name="text">
                     <string>移除</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item row="2" column="0">
                 <widget class="QGroupBox" name="groupBox_3">
                  <property name="title">
                   <string>选择目标地址后才可进行发送, 按住Ctrl+鼠标左键可多选目的地址</string>
                  </property>
                  <layout class="QGridLayout" name="gridLayout_8">
                   <item row="0" column="0">
                    <widget class="QListWidget" name="addrList">
                     <property name="selectionMode">
                      <enum>QAbstractItemView::ExtendedSelection</enum>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                </item>
                <item row="3" column="0">
                 <layout class="QHBoxLayout" name="horizontalLayout_3">
                  <item>
                   <widget class="QLabel" name="label">
                    <property name="text">
                     <string>仅发送主题号:</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLineEdit" name="includeEdit">
                    <property name="placeholderText">
                     <string>多主题号用英文逗号隔开, 例如：00C6,00C7</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="label_2">
                    <property name="text">
                     <string>排除主题号:</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLineEdit" name="excludeEdit">
                    <property name="placeholderText">
                     <string>多主题号用英文逗号隔开, 例如：00C6,00C7</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item row="4" column="0">
                 <layout class="QHBoxLayout" name="horizontalLayout_4">
                  <item>
                   <widget class="QLabel" name="label_3">
                    <property name="text">
                     <string>文件夹遍历顺序:</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QComboBox" name="orderCombo">
                    <item>
                     <property name="text">
                      <string>顺序</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>倒序</string>
                     </property>
                    </item>
                   </widget>
                  </item>
                  <item>
                   <widget class="QCheckBox" name="uniqueCheck">
                    <property name="text">
                     <string>每个主题号仅发送一次</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="label_4">
                    <property name="text">
                     <string>发送间隔(毫秒)：</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLineEdit" name="sendIntervalEdit"/>
                  </item>
                 </layout>
                </item>
               </layout>
              </widget>
              <widget class="QWidget" name="logTab">
               <attribute name="title">
                <string>日志显示</string>
               </attribute>
               <layout class="QGridLayout" name="gridLayout">
                <item row="0" column="0">
                 <widget class="QTextEdit" name="logView">
                  <property name="readOnly">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <item>
               <widget class="QPushButton" name="startButton">
                <property name="text">
                 <string>开始</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pauseButton">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>暂停(P)</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="stopButton">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>停止</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QLabel" name="statusLabel">
              <property name="text">
               <string>状态:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QProgressBar" name="progressBar">
              <property name="value">
               <number>0</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_2">
         <attribute name="title">
          <string>自定义数据发送</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_7">
          <item row="0" column="0">
           <layout class="QGridLayout" name="gridLayout_6">
            <item row="0" column="0">
             <widget class="QGroupBox" name="groupBox">
              <property name="title">
               <string>十六进制原始数据(每行一个数据帧)：</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_5">
               <item row="0" column="0">
                <layout class="QVBoxLayout" name="verticalLayout_6">
                 <item>
                  <layout class="QVBoxLayout" name="verticalLayout_3">
                   <item>
                    <widget class="QLabel" name="label_5">
                     <property name="text">
                      <string>格式参考(带不带分隔符均可)：</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QLineEdit" name="lineEdit">
                     <property name="text">
                      <string>80:21:00:C6:00:21:00:BA:F9:4C:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:01:00:00:00:00:02:8A:03</string>
                     </property>
                     <property name="readOnly">
                      <bool>true</bool>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                 <item>
                  <widget class="QTextEdit" name="textEdit">
                   <property name="placeholderText">
                    <string>请输入需要发送的原始数据...</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QGroupBox" name="groupBox_2">
              <property name="title">
               <string>发送日志：</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_4">
               <item row="0" column="0">
                <widget class="QTextEdit" name="textEdit_2"/>
               </item>
              </layout>
             </widget>
            </item>
            <item row="2" column="0">
             <layout class="QVBoxLayout" name="verticalLayout_4">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_7">
                <item>
                 <widget class="QLabel" name="label_7">
                  <property name="text">
                   <string>发送到(地址:端口号)：</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="addrEdit_2">
                  <property name="text">
                   <string>239.255.1.21:9221</string>
                  </property>
                  <property name="placeholderText">
                   <string>IP:Port</string>
                  </property>
                  <property name="clearButtonEnabled">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_6">
                <item>
                 <widget class="QLabel" name="label_6">
                  <property name="text">
                   <string>发送间隔(毫秒)：</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="sendIntervalEdit_2">
                  <property name="text">
                   <string>1000</string>
                  </property>
                  <property name="clearButtonEnabled">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QCheckBox" name="uniqueCheck_2">
                  <property name="text">
                   <string>是否循环发送</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_8">
                <item>
                 <widget class="QPushButton" name="startButton_2">
                  <property name="text">
                   <string>开始</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="pauseButton_2">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                  <property name="text">
                   <string>暂停(P)</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="stopButton_2">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                  <property name="text">
                   <string>停止</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="OrderSendWidget" name="tab_3">
         <attribute name="title">
          <string>指令数据发送</string>
         </attribute>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <action name="action_2">
   <property name="text">
    <string>科技蓝</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>OrderSendWidget</class>
   <extends>QWidget</extends>
   <header>orderSend/ordersendwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
